tests/dialects/test_vector.py

```diff
@@ -0,0 +1,79 @@
+from xdsl.dialects.builtin import i32, IntegerType, IndexType, VectorType
+from xdsl.dialects.memref import MemRefType
+from xdsl.dialects.vector import Load, Store
+from xdsl.ir import SSAValue, OpResult
+
+
+def test_vectorType():
+    vec1 = VectorType.from_type_and_list(i32)
+
+    assert vec1.get_num_dims() == 1
+    assert vec1.get_shape() == [1]
+    assert vec1.element_type is i32
+
+    vec2 = VectorType.from_type_and_list(i32, [3, 3, 3])
+
+    assert vec2.get_num_dims() == 3
+    assert vec2.get_shape() == [3, 3, 3]
+    assert vec2.element_type is i32
+
+    my_i32 = IntegerType.from_width(32)
+    vec3 = VectorType.from_params(my_i32)
+
+    assert vec3.get_num_dims() == 1
+    assert vec3.get_shape() == [1]
+    assert vec3.element_type is my_i32
+
+
+def test_vector_load_i32():
+    i32_memref_type = MemRefType.from_type_and_list(i32)
+    memref_ssa_value = SSAValue(i32_memref_type)
```

Comment:
The `SSAValue` class actually is an abstract class, so it is not supposed to be instantiated directly. Unfortunately, Python does not give functionality to really disallow this from happening.

Please use `OpResult` instead, this should work the exact same way.

I know this is used in other tests, but it is a mistake there as well, just one that wasn't fixed yet.